tests/unit/RequestBuilder/RecommendationRequestBuilderTest.php

```diff
@@ -108,4 +108,39 @@ public function shouldThrowExceptionWhenUserOfUserMergeDiffersFromSorting(): voi
         );
         $builder->build();
     }
+
+    /** @test */
+    public function shouldCheckSequenceOfUserIdsWhenMerging(): void
+    {
+        /**
+         * Should pass: ([interaction], [user merge], [recommendation]):
+         *              (A, A -> B, B)
+         */
+        $interactionCommand = Interaction::purchase('old-user-id', 'test-item-id');
+        $userMergeCommand = UserMerge::mergeFromSourceToTargetUser('old-user-id', 'new-user-id');
+        $recommendationsCommand = UserRecommendation::create('new-user-id', 5, 'scenario', 0.5, 3600);
+
+        $builder = new RecommendationRequestBuilder($recommendationsCommand);
+        $builder->setUserMerge($userMergeCommand);
+        $builder->setInteraction($interactionCommand);
+        $this->assertInstanceOf(Request::class, $builder->build());
+
+        /**
+         * Should fail: ([interaction], [user merge], [recommendation]):
+         *              (A, B -> A, A)
+         */
+        $interactionCommand = Interaction::purchase('old-user-id', 'test-item-id');
+        $userMergeCommand = UserMerge::mergeInto('old-user-id', 'new-user-id');
```

Comment:
The data naming is confusing - `mergeInto` params are `$targetUserId, $sourceUserIdToBeDeleted`. So call them `old-user-id` and `new-user-id` (while the later one is in fact the one to be deleted, ie. not the "new" user id) may lead to confusion imo.